fairroot/base/steer/FairRun.h

```diff
@@ -258,7 +260,10 @@ class FairRun : public TNamed
      *
      * \todo Should `fEvtHeader` be checked?
      */
-    UInt_t GetEvtHeaderRunId() const { return fEvtHeader->GetRunId(); }
+    [[deprecated("Use GetEventHeader()->GetRunId()")]] UInt_t GetEvtHeaderRunId() const
+    {
+        return fEvtHeader->GetRunId();
+    }
```

Comment:
Can we please also keep `GetEvtHeaderRunId()` for the same reason we keep `FillEventHeader()`?

Reply:
Hmmm, I still don't understand this style. 

`FairEventHeader` has other getters, like `GetInputFiledID`, `GetEventTime`, etc. Does that mean we should also define `GetEvtHeaderInputFieldID`, `GetEvtEventTime`, etc in `FairRun`? If so, we should do the same for `fSource`. And in the end, `FairRun` could be filled with more than hundreds of getters and setters.

Reply:
These are only intended for internal use in FairRun{,Ana,Sim} and not for public use. And so we only have "helpers" for exactly the few interactions with FairEventHeader, that happen often enough in FairRun\*, that it is worth to have an internal helper that simplifies internal use. And yes, for FairSource or FairSink, we could have those internal helpers, if they simplify things.

At least that was my idea when I introduced some of these.

Reply:
Sorry, I can't agree that those internal helper functions simplify things. There is nothing to simplify if there are just one or two lines of code. And on the contrary, it makes things more unreadable and less transparent. For example, from the name of `FillEventHeader()`, it's difficult to say to where this event header is filled and why it's related to a file source. Same goes for `GetEvtHeaderRunId()`. It means just one line of code `evtHeader->GetRunId()` and makes the code less transparent.

I think we should try to write a clean code. And a clean code includes no unnecessary interface or internal method. Sorry about wasting the time for this argument. Maybe a third person opinion could help us address this issue more quickly? @dennisklein

Reply:
Unfortunately, I don't have a good argument to add, which would favor one over the other :see_no_evil:

However, I would like to repeat the comment [I made earlier](https://github.com/FairRootGroup/FairRoot/pull/1254#discussion_r1237288780). If such a recommendation like `GetEventHeader()->GetRunId()` would be added, let's make some effort it is not going to be dangerous code. (I know there is already a good share of unsafe APIs and not-so-safe coding practices in FairRoot/ROOT, but if we never start to address them, the situation will never improve, even though it is only tiny steps)

Reply:
I agree ;D